packages/push/src/instruction/exec/mod.rs

```diff
@@ -0,0 +1,94 @@
+mod dup;
+mod ifelse;
+mod unless;
+mod when;
+
+use self::{dup::Dup, ifelse::IfElse, unless::Unless, when::When};
+use super::{instruction_error::PushInstructionError, Instruction, NumOpens, PushInstruction};
+use crate::{
+    error::InstructionResult,
+    push_vm::{program::PushProgram, HasStack},
+};
+
+#[derive(Debug, strum_macros::Display, Clone, Eq, PartialEq)]
+pub enum ExecInstruction {
+    Noop, /* Maybe use `Noop(())` instead of `Noop(Noop)` when we get around to this. See
+           * 2024-03-31 stream chat for more. */
+    Dup(Dup),
+    When(When),
+    Unless(Unless),
+    IfElse(IfElse),
+}
+
+impl ExecInstruction {
+    #[must_use]
```

Comment:
Do we just want to put the attribute on the enum directly instead? Then all values of type ExecInstruction must be used, and we can remove the annotation here

Reply:
That seemed like a great idea, but it doesn't work. Putting a `#[must_use]` on an `impl` has no effect, and you still get the Clippy suggestions on each of the methods. Nice try, though!

Reply:
not on the impl, on the enum directly, like 

```rs
#[must_use]
enum XYZ {}
```

Reply:
That worked much better – thanks!